templates/account.php

```diff
@@ -590,14 +590,100 @@ class="fs-tag fs-<?php echo $fs->can_use_premium_code() ? 'success' : 'warn' ?>"
 										</tr>
 										</thead>
 										<tbody>
-										<?php $odd = true;
+										<?php
+                                            $odd = true;
+
+                                            $installed_addons_ids_map = array_flip( $installed_addons_ids );
+
+                                            $fs_options = FS_Options::instance( WP_FS__ACCOUNTS_OPTION_NAME );
+
+                                            $sites = $fs_options->get_option( 'sites', array() );
+
+                                            $plugins_data = null;
+                                            $all_plans    = null;
+                                            $licenses     = null;
+
 											foreach ( $addons_to_show as $addon_id ) {
+                                                $addon              = $fs->get_addon( $addon_id );
+                                                $slug               = $addon->slug;
+											    $is_addon_connected = false;
+											    $addon_info         = array();
+
+                                                if ( isset( $sites[ $slug ] ) ) {
+                                                    $site   = $sites[ $slug ];
+                                                    $plugin = FS_Plugin_Manager::instance( $addon_id )->get();

```

Comment:
@fajardoleo I feel that there's a confusion in the var names. For example, `$fs->get_addon( $addon_id )` also returns an `FS_Plugin` instance just like `FS_Plugin_Manager::instance( $addon_id )->get()`, so in the case of an active add-on, both values will be identical but the vars are inconsistent (`$addon` vs `$plugin). You also added `foreach`s that that use `$plan`, `$license`, and `$subscription` for the internal loop vars which will override the template's global values (not a good practice). I think that it worth trying to move all this new logic to a helper function.